src/dmd/globals.d

```diff
@@ -469,6 +470,24 @@ nothrow:
                FileName.equals(filename, loc.filename);
     }
 
+    extern (D) bool opEquals(Loc loc) const pure
+    {
+        return charnum == loc.charnum &&
+               linnum == loc.linnum &&
+               FileName.equals(filename, loc.filename);
+    }
+
+    extern (D) hash_t toHash() const pure
+    {
+        import dmd.root.hash : mixHash;
+
+        auto hash = hashOf(linnum);
+        hash = mixHash(hash, hashOf(charnum));
+        if (filename !is null)
+            hash = mixHash(hash, hashOf(filename[0 .. strlen(filename)]));
```

Comment:
Please tell me there's something like `mixHash()` in druntime and an existing helper function converting a C string to a D one.

Reply:
Custom hashing is necessary because of mixins, e.g.:
```
test/runnable/test18880.d-mixin-7 (0x558d3e709930)
test/runnable/test18880.d-mixin-7 (0x558d3e70a120)
test/runnable/test18880.d-mixin-7 (0x558d3e70a3f0)
```

Btw, 2.081.1 isn't able to compile master, that's pretty bad.